ledger-core/v2/virtual/integration/delegated_finished_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package small
+
+import (
+	"testing"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/stretchr/testify/require"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/gen"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/instrumentation/inslogger"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/integration/utils"
+)
+
+// 1. Send CallRequest
+// 2. Since there is no state SM send VSateRequest
+// 3. We catch VSateRequest. Change pulse. Send VStateReport
```

Comment:
Not sure if it's right place to change pulse here.
I think it should be changed inside contract execution

Reply:
mock executor and change pulse inside of this mock